puppet/modules/jenkins_job_builder/manifests/init.pp

```diff
@@ -0,0 +1,60 @@
+# Class to install, configure, and maintain JJB, as well
+# as to deploy the actual jobs to jenkins
+#
+# Loosely based on
+# https://git.openstack.org/cgit/openstack-infra/puppet-jenkins/tree/manifests/job_builder.pp
+# and should probably be kept up to date from there
+#
+class jenkins_job_builder (
+  $url = '',
+  $username = '',
+  $password = '',
+  $config_dir = '',
+) {
+
+  if ! defined(Package['jenkins-job-builder']) {
+    package { 'jenkins-job-builder':
+      ensure   => present,
+      provider => 'pip',
+    }
+  }
+
+  file { '/etc/jenkins_jobs':
+    ensure => directory,
+  }
+
+  file { '/etc/jenkins_jobs/config':
+    ensure  => directory,
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0755',
+    recurse => true,
+    purge   => true,
+    force   => true,
+    source  => $config_dir,
+    require => File['/etc/jenkins_jobs'],
+    notify  => Exec['jenkins_jobs_update'],
+  }
+
+  exec { 'jenkins_jobs_update':
+    #command    => 'jenkins-jobs update --delete-old /etc/jenkins_jobs/config', #danger!
+    #command    => 'jenkins-jobs update /etc/jenkins_jobs/config',              #op
+    command     => 'jenkins-jobs test /etc/jenkins_jobs/config > /tmp/jjb.xml', #no-op
+    timeout     => '600',
+    path        => '/bin:/usr/bin:/usr/local/bin',
+    refreshonly => true,
+    require     => [
+      File['/etc/jenkins_jobs/jenkins_jobs.ini'],
+      Package['jenkins-job-builder'],
+    ],
+  }
+
+# TODO: We should put in  notify Exec['jenkins_jobs_update']
+#       at some point, but that still has some problems.
```

Comment:
Thats copied from the openstack code. I left it in so I could see when they fixed it, if we ever update our code from theirs...